### PR TITLE
MTSRE-348 | Add: addon metadata addOnParameters validator

### PR DESCRIPTION
**Pull request by yashvardhan-kukreja**
Signed-off-by: Yashvardhan Kukreja <ykukreja@redhat.com>

### CODE REVIEW

**Review by sugarraysam — COMMENTED** [review]
lgtm, thank you for this MR. Again please lets merge and rebase #34 . And add some `failureMsgs` .

**Comment by yashvardhan-kukreja** [comment]
please merge #38 before merging this PR

**Review by sugarraysam — APPROVED** [review]

**Comment by sugarraysam** [comment]
@yashvardhan-kukreja please rebase

File: pkg/validators/test_am0005.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+func (val TestAM0005) Name() string {
+	return AM0005.Name
+}
+
+func (val TestAM0005) Run(mb utils.MetaBundle) (bool, string, error) {
+	return AM0005.Runner(mb)
+}
+
+func (val TestAM0005) SucceedingCandidates() []utils.MetaBundle {
```
Comment (sugarraysam) [inline pkg/validators/test_am0005.go:23]: Can we also add the `testutils.DefaultSucceedingCandidates()`

**Review by sugarraysam — APPROVED** [review]
lgtm

please consider creating wiki page: https://github.com/mt-sre/addon-metadata-operator/wiki/AM0005

**Comment by yashvardhan-kukreja** [comment]
> please consider creating wiki page: https://github.com/mt-sre/addon-metadata-operator/wiki/AM0005

Done @sugarraysam

**Comment by sugarraysam** [comment]
lgtm

awesome, one last thing, please make sure to test this against prod addons in managed-tenants. https://gitlab.cee.redhat.com/service/managed-tenants-sops/-/blob/main/MT-SRE/sops/AMO-validators.md before merging

And since this code will replace `tasks/check/40_test_addon_parameters.py`, when integrating let's make sure to archive this task so CI doesn't run it twice -> `tasks/archived/50_test_addon_parameters.py`.

Thank you !

**Comment by yashvardhan-kukreja** [comment]
> awesome, one last thing, please make sure to test this against prod addons in managed-tenants.

@sugarraysam  done this! Works well :)

**Comment by yashvardhan-kukreja** [comment]
> And since this code will replace tasks/check/40_test_addon_parameters.py, when integrating let's make sure to archive this task so CI doesn't run it twice -> tasks/archived/50_test_addon_parameters.py.

Sure, I'll raise an MR for the above but let's block that MR until managed-tenants repo doesn't start making use of the latest mtcli consisting of the above validator.

**Comment by sugarraysam** [comment]
@yashvardhan-kukreja are we ready to merge this?

**Comment by yashvardhan-kukreja** [comment]
yep :)

**Pull request merged**